File: ALODWebUtility/Common/ChangeSetDetails.vb
Imports ALOD.Data

Namespace Common

    Public Class ChangeSetDetails

        Private _actionDate As New Nullable(Of Date)
        Private _firstName As String
        Private _lastName As String
        Private _logId As Integer
        Private _rank As String
        Private _userId As Integer

        Public Property ActionDate() As Nullable(Of Date)
            Get
                Return _actionDate
            End Get
            Set(ByVal value As Nullable(Of Date))
                _actionDate = value
            End Set
        End Property

        Public Property FirstName() As String
            Get
                Return _firstName
            End Get
            Set(ByVal value As String)
                _firstName = value
            End Set
        End Property

        Public ReadOnly Property FullNameRank() As String
            Get
                If (Not IsNothing(_rank) AndAlso Rank.ToLower() <> "civ") Then
                    Return _rank + " " + _lastName + ", " + _firstName
                End If

                If (Not IsNothing(_lastName) AndAlso Not IsNothing(_firstName)) Then
                    Return _lastName + ", " + _firstName
                End If

                Return Nothing

            End Get
        End Property

        Public Property LastName() As String
            Get
                Return _lastName
            End Get
            Set(ByVal value As String)
                _lastName = value
            End Set
        End Property

        Public Property LogId() As Integer
            Get
                Return _logId
            End Get
            Set(ByVal value As Integer)
                _logId = value
            End Set
        End Property

        Public Property Rank() As String
            Get
                Return _rank
            End Get
            Set(ByVal value As String)
                _rank = value
            End Set
        End Property

        Public Property UserId() As Integer
            Get
                Return _userId
            End Get
            Set(ByVal value As Integer)
                _userId = value
            End Set
        End Property

        Public Function GetLastChange(ByVal id As Integer) As ChangeSetDetails
            Dim adapter As New SqlDataStore()
            adapter.ExecuteReader(AddressOf LastChagedReader, "core_log_sp_GetLastChange", id)
            Return Me
        End Function

        Protected Sub LastChagedReader(ByVal adapter As SqlDataStore, ByVal reader As System.Data.IDataReader)
            Me.LogId = adapter.GetInt32(reader, 0)
            Me.ActionDate = adapter.GetNullableDate(reader, 1, Nothing)
            Me.UserId = adapter.GetInt32(reader, 2)
            Me.LastName = adapter.GetString(reader, 3)
            Me.FirstName = adapter.GetString(reader, 4)
            Me.Rank = adapter.GetString(reader, 5)
        End Sub

    End Class

End Namespace

File: ALODWebUtility/Common/ChangeSetDetails.cs
using System;
using System.Data;
using ALOD.Data;

namespace ALODWebUtility.Common
{
    public class ChangeSetDetails
    {
        private DateTime? _actionDate;
        private string _firstName;
        private string _lastName;
        private int _logId;
        private string _rank;
        private int _userId;

        public DateTime? ActionDate
        {
            get { return _actionDate; }
            set { _actionDate = value; }
        }

        public string FirstName
        {
            get { return _firstName; }
            set { _firstName = value; }
        }

        public string FullNameRank
        {
            get
            {
                if (!string.IsNullOrEmpty(_rank) && _rank.ToLower() != "civ")
                {
                    return _rank + " " + _lastName + ", " + _firstName;
                }

                if (!string.IsNullOrEmpty(_lastName) && !string.IsNullOrEmpty(_firstName))
                {
                    return _lastName + ", " + _firstName;
                }

                return null;
            }
        }

        public string LastName
        {
            get { return _lastName; }
            set { _lastName = value; }
        }

        public int LogId
        {
            get { return _logId; }
            set { _logId = value; }
        }

        public string Rank
        {
            get { return _rank; }
            set { _rank = value; }
        }

        public int UserId
        {
            get { return _userId; }
            set { _userId = value; }
        }

        public ChangeSetDetails GetLastChange(int id)
        {
            SqlDataStore adapter = new SqlDataStore();
            adapter.ExecuteReader(LastChagedReader, "core_log_sp_GetLastChange", id);
            return this;
        }

        protected void LastChagedReader(SqlDataStore adapter, IDataReader reader)
        {
            this.LogId = adapter.GetInt32(reader, 0);
            this.ActionDate = adapter.GetNullableDate(reader, 1, null);
            this.UserId = adapter.GetInt32(reader, 2);
            this.LastName = adapter.GetString(reader, 3);
            this.FirstName = adapter.GetString(reader, 4);
            this.Rank = adapter.GetString(reader, 5);
        }
    }
}
